packages/memory-profiler/src/unpack.ts

```diff
@@ -0,0 +1,177 @@
+import * as path from 'path';
+
+import { ComponentSourceMaps } from '@fitbit/app-package';
+import * as cbor from 'cbor';
+import * as t from 'io-ts';
+import { failure } from 'io-ts/lib/PathReporter';
+import * as lodash from 'lodash';
+import { SourceMapConsumer, RawSourceMap } from 'source-map';
+
+import { Edge, HeapSnapshot, Node, NodeTypes, RawNode, RawEdge } from './types';
+
+type RawNodeMap = Record<number, RawNode>;
+type NodeMap = Record<number, Node>;
+
+// FIXME: copied from sdk-cli
+const mapValues = <T, U>(
+  obj: { [s: string]: T },
+  mapper: (value: T, key: string, index: number) => Promise<U> | U,
+) =>
+  Promise.all(
+    Object.entries(obj).map(
+      async ([key, value], index) => ({ [key]: await mapper(value, key, index) })),
+  ).then(entries => entries.reduce((a, b) => ({ ...a, ...b }), {}));
+
+function validateOrThrow<A, O, I>(type: t.Type<A, O, I>, data: I): A {
+  return type.decode(data).getOrElseL((errors) => {
+    throw new Error(failure(errors).join('\n'));
+  });
+}
+
+function normalizeRawNode({ id, type, size, repr }: RawNode, rawNodes: RawNodeMap): Node {
+  if (Array.isArray(repr)) {
+    const [source, line, column, name] = repr;
+    return {
+      id,
+      type,
+      size,
+      position: {
+        line,
+        column,
+        name: name || undefined,
+        source: typeof source === 'string' ? source : rawNodes[source].repr,
+      },
+    };
+  }
+
+  if (repr !== null) {
```

Comment:
i'm not that familiar with `io-ts`, but is there a way to avoid using `null` and just use `undefined` everywhere? Or are we the ones that unpack the snapshot values into `null`?

Reply:
No, `null` is encoded into the snapshot in FW.